outlines/caching.py

```diff
@@ -17,6 +21,30 @@ def cache_fn(fn: Callable):
     return cache_fn
 
 
+def hash_data(*data) -> str:
+    """Pickles and hashes all the data passed to it as args."""
+    result = hashlib.md5()  # nosec B303
+    for datum in data:
+        if isinstance(datum, torch.Tensor):
+            datum = datum.cpu().numpy()
+        result.update(cloudpickle.dumps(datum))
+    return result.hexdigest()
+
+def diskcache(cache_key_args_func):
```

Comment:
We should have a docstring here that explains what this decorator does

Reply:
Also, do we really need to pass a function? Isn't it enough to just pass the cache keys?

Reply:
Since we are modifying the (cached) function arguments before passing them to the decorator, hence we need a separate function here.